dask/array/ufunc.py

```diff
@@ -109,9 +109,13 @@ def __repr__(self):
         return repr(self._ufunc)
 
     def __call__(self, *args, **kwargs):
-        dsk = [arg for arg in args if hasattr(arg, '_elemwise')]
-        if len(dsk) > 0:
-            return dsk[0]._elemwise(self._ufunc, *args, **kwargs)
+        dsks = [arg for arg in args if hasattr(arg, '_elemwise')]
+        if len(dsks) > 0:
+            for dsk in dsks:
+                result = dsk._elemwise(self._ufunc, *args, **kwargs)
+                if type(result) != type(NotImplemented):
+                    return result
+            return NotImplemented
```

Comment:
Calling a ufunc shouldn't ever return NotImplemented. Instead, if every implementation returns `NotImplemented`  you should raise `TypeError` (like how Python handles arithmetic).